cli/src/commands/sign.rs

```diff
@@ -0,0 +1,161 @@
+    )]
+    revisions: Vec<RevisionArg>,
+}
+
+pub fn cmd_sign(ui: &mut Ui, command: &CommandHelper, args: &SignArgs) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+
+    if command.settings().signing_backend()?.is_none() {
+        return Err(user_error_with_hint(
+            "No signing backend configured.",
+            "For configuring a signing backend, see https://jj-vcs.github.io/jj/latest/config/#commit-signing",
+        ));
+    }
+
+    let is_authored_by_me =
+        |commit: &Commit| -> bool { commit.author().email == command.settings().user_email() };
+    let is_signed_by_me =
+        |commit: &Commit| -> bool { commit.is_signed() && is_authored_by_me(commit) };
+
+    let to_sign: IndexSet<Commit> = workspace_command
+        .parse_union_revsets(ui, &args.revisions)?
+        .evaluate_to_commits()?
+        .filter_ok(|commit| !is_signed_by_me(commit))
+        .try_collect()?;
+
+    workspace_command.check_rewritable(to_sign.iter().ids())?;
+
+    let mut tx = workspace_command.start_transaction();
+
+    let mut signed_commits = vec![];
+    let mut foreign_commits = vec![];
+    let mut num_reparented = 0;
+
+    tx.repo_mut().transform_descendants(
+        to_sign.iter().ids().cloned().collect_vec(),
+        |rewriter| {
+            let old_commit = rewriter.old_commit().clone();
+            let commit_builder = rewriter.reparent();
+
+            if to_sign.contains(&old_commit) {
+                let new_commit = commit_builder
+                    .set_sign_key(args.key.clone())
+                    .set_sign_behavior(SignBehavior::Own)
+                    .write()?;
+
+                if !is_authored_by_me(&old_commit) {
+                    foreign_commits.push(new_commit.clone());
+                }
+
+                signed_commits.push(new_commit);
+            } else {
+                commit_builder.write()?;
+                num_reparented += 1;
+            }
+
+            Ok(())
+        },
+    )?;
+
+    if let Some(mut formatter) = ui.status_formatter() {
+        match &*foreign_commits {
+            [] => {}
+            [commit] => {
+                writeln!(ui.warning_default(), "Signed 1 commit not authored by you")?;
+                write!(formatter, "  ")?;
+                tx.base_workspace_helper()
+                    .write_commit_summary(formatter.as_mut(), commit)?;
+                writeln!(ui.status())?;
+            }
+            commits => {
+                let template = tx.base_workspace_helper().commit_summary_template();
+                writeln!(
+                    ui.warning_default(),
+                    "Signed {} commits not authored by you",
+                    commits.len()
+                )?;
+                for commit in commits {
+                    write!(formatter, "  ")?;
+                    template.format(commit, formatter.as_mut())?;
+                    writeln!(formatter)?;
+                }
```

Comment:
Since these outputs are mostly identical, I would combine them. We usually use plural form.

Reply:
All right, I wasn't really sure how "perfect" we want the phrasing to be.
Happy to go with the plural form then.